projects/igniteui-angular/src/lib/grids/common/pipes.ts

```diff
@@ -355,3 +355,39 @@ export class IgxColumnFormatterPipe implements PipeTransform {
         return formatter(value);
     }
 }
+
+@Pipe({
+    name: 'gridAddRow',
+    pure: true
+})
+export class IgxGridAddRowPipe implements PipeTransform {
+
+    constructor(private gridAPI: GridBaseAPIService<IgxGridBaseDirective & GridType>) { }
+
+    transform(collection: any, parentID: number, enableAddRow: boolean, cancelAddMode: boolean, pipeTrigger: number) {
+        const grid = this.gridAPI.grid;
+        if (!enableAddRow) {
```

Comment:
don't pass grid props as pipe state, you can simply pass the grid. it would also be wrong as you can use the API to start adding a row without the need of a UI flag